socorro/processor/mozilla_transform_rules.py

```diff
@@ -778,45 +864,49 @@ def action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
         except ValueError:
             build_id = None
         release_channel = processed_crash.get('release_channel').strip()
-        version = processed_crash.get('version').strip()
+        version = processed_crash.get('version', '').strip()
+        uuid = raw_crash.get('uuid', None)
 
-        # If we're missing one of the magic ingredients, then there's nothing
-        # to do
+        # Only run if we've got all the things we need
         if product and build_id and release_channel and product in self.BUILDHUB_PRODUCTS:
             # Convert the build_id to a str for lookups
             build_id = str(build_id)
 
-            try:
-                real_version = self._get_real_version(product, build_id, release_channel)
+            real_version = self.get_version(uuid, product, build_id, release_channel, version)
 
-                # If we got a real version, toss that in and we're done
-                if real_version:
-                    processed_crash['version'] = real_version
-                    return
-
-                # We didn't get a version from Buildhub, but this might be a
-                # "final beta" which Buildhub has an entry for in the release
-                # channel. So we check Buildhub asking about the release
-                # channel and get back an rc version.
-                if version and release_channel == 'beta' and self.is_final_beta(version):
-                    real_version = self._get_real_version(product, build_id, 'release')
-
-                    if real_version:
-                        processed_crash['version'] = real_version
-                        return
-
-            except RequestException as exc:
-                processor_meta.processor_notes.append('could not connect to Buildhub')
+            # Now look it up with the alternative method using the
+            # crashstats_productversion table and compare
+            maybe_version = None
+            try:
+                maybe_version = self._get_real_version_db2(product, release_channel, build_id)
+                if real_version != maybe_version:
+                    self.config.logger.info(
+                        'betaversionrule: got different versions: %r %r vs. %r',
+                        raw_crash.get('uuid', None),
+                        real_version,
+                        maybe_version
+                    )
+                    self.metrics.incr('lookup_alternate', 1, tags=['outcome:different'])
+                else:
+                    self.metrics.incr('lookup_alternate', 1, tags=['outcome:same'])
```

Comment:
We can run this on stage over the weekend and graph this and have a really good idea of how it fares in the real world.